xarray/core/dataarray.py

```diff
@@ -1377,13 +1378,15 @@ def func(self, *args, **kwargs):
         return func
 
     @staticmethod
-    def _binary_op(f, reflexive=False, join='inner', **ignored_kwargs):
+    def _binary_op(f, reflexive=False, join=None, **ignored_kwargs):
         @functools.wraps(f)
         def func(self, other):
             if isinstance(other, (Dataset, groupby.GroupBy)):
                 return NotImplemented
             if hasattr(other, 'indexes'):
-                self, other = align(self, other, join=join, copy=False)
+                # if user does not specify join, default to OPTIONS['join']
+                how_to_join = join or OPTIONS['join']
```

Comment:
Yeah I think `arithmetic_join` or `compute_join` is probably the way to go. We will probably also use this when/if we ever add ternary operations, too.